### PR TITLE
docs(ripple): update documentation

### PR DESCRIPTION
**Pull request by DevVersion**
* Improves the documentation of the `matRipple` attribute directive.

References #7543.

### CODE REVIEW

File: src/lib/core/ripple/ripple.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+The `matRipple` attribute directive defines an area in which a ripple animates on user interaction.
+
+```html
+<div matRipple [matRippleColor]="myColor">
+  <ng-content></ng-content>
+</div>
+```
+
+By default, a ripple is activated when the host element of the `matRipple` directive receives
+mouse or touch events. Once element is being pressed, a ripple starts fading in from the point
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: "Once element is being pressed..." ->
"Upon being pressed, a ripple will begin fading in from the point of contact, radiating to cover the host element"

File: src/lib/core/ripple/ripple.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+of contact and expands circularly to cover the full host element.
+
+As long as the user does not release the element, the ripples will not fade out. Releasing the
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: "Each ripple will fade out only upon release of the mouse or touch"

File: src/lib/core/ripple/ripple.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+element, will make the present ripples fade out.
+
+Ripples can also be triggered programmatically by getting a reference to the MatRipple directive
+and calling its `launch` method.
+
+
+### Ripple trigger
+
+As mentioned previously, by default ripples will fade in on interaction with the directive's
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: Omit "As mentioned previously"

File: src/lib/core/ripple/ripple.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+host element.
+
+In some situations, developers may want to show ripples on interaction with *another* element,
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: Combine with previous paragraph

Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: "another" -> "some other"

File: src/lib/core/ripple/ripple.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+### Ripple trigger
+
+As mentioned previously, by default ripples will fade in on interaction with the directive's
+host element.
+
+In some situations, developers may want to show ripples on interaction with *another* element, 
+but still want to have the ripples placed in another location.
+
+This can be done by specifying the `matRippleTrigger` option that expects a reference to a
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: Combine with previous paragraph

Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: "an HTMLElement"

File: src/lib/core/ripple/ripple.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+@NgModule({
+  providers: [
+    {provide: MAT_RIPPLE_GLOBAL_OPTIONS, useValue: globalRippleConfig} 
+  ]
+})
+```
+
+All currently available global options are shown here:
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: Could this just be captured by API docs?

Comment (DevVersion) [reply src/lib/core/ripple/ripple.md]: Interfaces are not being displayed in the docs yet. We might be able to change the interface to a class, but I'd rather would like to setup Dgeni to show interfaces (in a follow-up)

File: src/lib/core/ripple/ripple.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
+# matRipple
```
Comment (jelbourn) [inline src/lib/core/ripple/ripple.md]: Omit the top `#matRipple` header (it is added by the docs site)

**Pull request merged**